### PR TITLE
Fix #126: Create Composer script to update version.

### LINKED ISSUE #126: Update assets version automatically
Currently we are forced to update manually here: https://github.com/jolicode/secret-santa/blob/master/config/packages/framework.yaml#L16

It could be easily replaced by a hash of the assets directory.

### PR DESCRIPTION
**Pull request by mookman288**
I wasn't sure if `public` was the correct asset directory, but it's easily changeable in the Composer script. I was unable to auto-load the Yaml libraries in Symfony because of a php version mis-match, but I did test it with a manual load of the Yaml libraries and it worked flawlessly. Hope this meets your needs.

### CODE REVIEW

**Comment by pyrech** [comment]
Hi, thanks a lot for your PR :slightly_smiling_face: 

I'm not a big fan of this Composer script that update a versioned YAML file. As Symfony supports configuration in PHP for the DIC, I would prefer a small PHP config file that set the asset version directly at container compilation time.

Something like this:

```php
<?php
// config/packages/assets_version.php

$assetVersion = ...

$container->loadFromExtension('framework', [
    'assets' => [
        'version' => $assetVersion,
    ],
]);
```

Would you mind to move your logic in such a config file?

Thanks a lot!

File: config/packages/assets_version.php [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    *
+    * (c) JoliCode <coucou@jolicode.com>
+    *
+    * For the full copyright and license information, please view the LICENSE
+    * file that was distributed with this source code.
+    */
+
+	$container->loadFromExtension('framework', [
+		'assets' => [
+			'version' => function() {
```
Comment (lyrixx) [inline config/packages/assets_version.php]: You should not pass a callable, but the real value instead

File: config/packages/assets_version.php [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+				$rootDirectory = sprintf("%s/../../", __DIR__);
+				$assetsDirectory = realpath(sprintf("%s/public", $rootDirectory));
+
+				$rdi = new \RecursiveDirectoryIterator($assetsDirectory);
+				$rii = new \RecursiveIteratorIterator($rdi);
```
Comment (lyrixx) [inline config/packages/assets_version.php]: Could you use the Symfony Finder instead ?

File: config/packages/assets_version.php [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+				$rawHash = '';
+
+				foreach($rii as $file) {
+					if (substr($file, -1) === '.' || substr($file, -1) === '..') continue;
+
+					$rawHash .= hash_file('sha256', $file);
```
Comment (lyrixx) [inline config/packages/assets_version.php]: Could you update the code to use `hash_init()`, `hash_update_file()`, and `hash_final()`.
Thanks

**Comment by lyrixx** [comment]
I did not test it, but it should be something like:
```

$files = (new Symfony\Component\Finder\Finder())
    ->in(__DIR__.'/../../..')
    // may be we should only select few extensions...
    ->sortByName()
    ->files()
;

$hashCtx = hash_init();

foreach ($files as $file) {
    hash_update_file($hashCtx, $file);
}

$container->loadFromExtension('framework', [
    'assets' => [
        'version' => hash_final($hashCtx),
    ]
]);
```
---
Note: another solution is to make an hash of the `packages.json` file

**Comment by pyrech** [comment]
Hello @mookman288, do you think you can make the required changes in the coming days? If you need help, do not hesitate to ask here :wink:

**Comment by mookman288** [comment]
@pyrech still working on it. Thanks.

**Comment by pyrech** [comment]
Didn't see you updated the PR, sorry for the delay.

It works perfectly, thanks a lot @mookman288

**Pull request merged**